zion/src/main/scala/edu/uci/ics/cloudberry/zion/actor/DataStoreManager.scala

```diff
@@ -152,6 +112,56 @@ class DataStoreManager(metaDataset: String,
   //persistent metadata periodically
   context.system.scheduler.schedule(config.ViewMetaFlushInterval, config.ViewMetaFlushInterval, self, FlushMeta)
 
+  private def registerNewTable(sender: ActorRef, registerTable: Register): Unit = {
+    val dataSetName = registerTable.dataset
+    val dataSetRawSchema = registerTable.schema
+
+    if(!metaData.contains(dataSetName)){
+
+      val primaryKey = dataSetRawSchema.primaryKey.map(dataSetRawSchema.getField(_).get)
+      val timeField = dataSetRawSchema.getField(dataSetRawSchema.timeField) match {
+        case Some(f)
+          if f.isInstanceOf[TimeField] => f.asInstanceOf[TimeField]
+        case None
+          if dataSetRawSchema.timeField.isEmpty => throw new QueryParsingException(s"Time field is not specified for ${dataSetRawSchema.typeName}.")
+        case _ =>
+          throw new QueryParsingException(s"${dataSetRawSchema.timeField} is not a valid time field.")
+      }
+      val resolvedSchema = Schema(dataSetRawSchema.typeName, dataSetRawSchema.dimension, dataSetRawSchema.measurement, primaryKey, timeField)
+
+      //TODO: Send query to get actual information when a dataset is registered.
+      val currentDateTime = new DateTime()
+      val fakeStats = Stats(currentDateTime, currentDateTime, currentDateTime, 1000000)
+
+      val fakeStartDate = new DateTime(2005, 3, 26, 12, 0, 0, 0)
+      val fakeInterval = new Interval(fakeStartDate, currentDateTime)
+
+      val registerDataSetInfo = DataSetInfo(dataSetName, None, resolvedSchema, fakeInterval, fakeStats)
+      metaData.put(dataSetName, registerDataSetInfo)
+      flushMetaData()
+
+      sender ! FrontEndRequestReceipt(true, "Register Finished: dataset " + dataSetName + " has successfully registered.")
+    }
+    else{
+      sender ! FrontEndRequestReceipt(false, "Register Denied: another dataset with the same name " + dataSetName + " has already existed in database.")
```

Comment:
I think simpler response is better: `Register Denied: dataset $dataSetName has already existed`.